examples/tools/android_native_run/main.cpp

```
@@ -0,0 +1,113 @@
+#include "JVM.h"
+#include "classpath.h"
+#include "options.h"
+#include "path.h"
+#include "Other_libraries.h"
+
+#include <csignal>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <dlfcn.h>
+#include <fstream>
+#include <getopt.h>
+#include <iostream>
+#include <jni.h>
+#include <sstream>
+#include <stdarg.h>
+#include <string>
+#include <unistd.h>
+#include <utility>
+#include <vector>
+
+#include <LIEF/LIEF.hpp>
+#include <QBDL/Engine.hpp>
+#include <QBDL/engines/Native.hpp>
+#include <QBDL/loaders/ELF.hpp>
+
+using namespace QBDL;
+using namespace std;
+
+
+
+namespace {
+
+struct FinalTargetSystem : public Engines::Native::TargetSystem {
+  void* last_lib;
+
+  FinalTargetSystem(TargetMemory &mem, vector<string> const &other_libraries) : Engines::Native::TargetSystem(mem)
+  {
+    for (auto const &lib : other_libraries)
+    {
+      last_lib = dlopen(lib.c_str(), RTLD_NOW);
+      if (!last_lib)
+      {
+        cerr << "can't open library " << lib << endl;
+        exit(1);
+      }
+    }
+
+  }
+  
+  
+  uint64_t symlink(Loader &loader, const LIEF::Symbol &sym) override {
+    printf("%s\n", sym.name().c_str());
+    dlerror();
```

The error here doesn't seem to be used; I think there are two options here:
  - do not use `dlerror`, and silently ignore when symbols are actually undefined (which may lead to undocumented segfault at runtime)
  - actually do `dlsym` then `dlerror`, emit a warning if a symbols is not found, and return the result of dlsym anyway. The program would still segfault if an unloaded symbol is executed, but at least there is a message that something may go wrong.

---

Done